src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -810,128 +916,17 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
-                if (relationship.TargetFields is not null)
-                {
-                    GetFieldsNotBackedByColumnsInDB(
-                        fields: relationship.TargetFields,
-                        invalidColumns: invalidColumns,
-                        entityName: relationship.TargetEntity,
-                        sqlMetadataProvider: sqlMetadataProvider);
-
-                    if (invalidColumns.Count > 0)
-                    {
-                        HandleOrRecordException(new DataApiBuilderException(
-                            message: $"Entity: {entityName} has a relationship: {relationshipName} with target fields: {string.Join(",", invalidColumns)} that " +
-                                $"do not exist as columns in entity: {relationship.TargetEntity}.",
-                            statusCode: HttpStatusCode.ServiceUnavailable,
-                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
-                    }
-                }
-
-                // Linking object exists and we therefore have a many-many relationship. Validation here differs from one-many and many-one in that
-                // the source and target fields are now only indirectly related through the linking object. Therefore, it is the source and linkingSource
-                // along with the target and linkingTarget fields that must match, respectively. Source and linkingSource fields provide the relationship
-                // from the source entity to the linkingObject while target and linkingTarget fields provide the relationship from the target entity to the
-                // linkingObject.
-                // see: https://learn.microsoft.com/en-us/azure/data-api-builder/relationships#many-to-many-relationship
-                if (!string.IsNullOrWhiteSpace(relationship.LinkingObject))
-                {
-                    ValidateFieldsAndAssociatedLinkingFields(
-                        fields: relationship.SourceFields,
-                        linkingFields: relationship.LinkingSourceFields,
-                        fieldType: "source",
-                        entityName: entityName,
-                        relationshipName: relationshipName);
-                    ValidateFieldsAndAssociatedLinkingFields(
-                        fields: relationship.TargetFields,
-                        linkingFields: relationship.LinkingTargetFields,
-                        fieldType: "target",
-                        entityName: entityName,
-                        relationshipName: relationshipName);
-                }
+                ValidateSourceAndTargetFieldsAsBackingColumns(
+                    entityName: entityName,
+                    relationshipName: relationshipName,
+                    relationship: relationship,
+                    sqlMetadataProvider: sqlMetadataProvider,
+                    invalidColumns: invalidColumns);
```

Comment:
Need not have `invalidColumns` here. Just referred to in the downstream.

Reply:
we dont want to create and return a new list because we dont have any guarantees about garbage collection, so we need to pass a reference to the list we created before the loop.